dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserService.java

```diff
@@ -464,6 +465,18 @@ static boolean hasTwoFactorSecretForApproval( User user )
      */
     List<User> getUsersWithAuthority( String authority );
 
+    /**
+     * Use this method instead of {@link #createUserDetails(User)} if no
+     * {@link User} instance is available or if the one available is not fully
+     * loaded or connected to a session.
+     *
+     * @see #createUserDetails(User)
+     * @param userUid UID of the {@link CurrentUserDetails} to create
+     * @return the implementation object
+     */
+    CurrentUserDetails createUserDetails( String userUid )
```

Comment:
FYI: The reason I added this is because the `User` instance usually is not attached to a session in a worker thread which then fails when roles are accessed to compute the details. So it is better to load the user within this method so it is attached to the session opened by the service method.